lib/ModelMixins/TimeFilterMixin.ts

```diff
@@ -1,37 +1,37 @@
-import Constructor from "../Core/Constructor";
-import Model from "../Models/Model";
-import TimeFilterTraits, {
-  TimeFilterCoordinates
-} from "../Traits/TimeFilterTraits";
-import DiscretelyTimeVaryingTraits from "../Traits/DiscretelyTimeVaryingTraits";
-import { ProviderCoordsMap, ProviderCoords } from "../Map/PickedFeatures";
-import Entity from "terriajs-cesium/Source/DataSources/Entity";
-import createStratumInstance from "../Models/createStratumInstance";
-import DiscreteTimeTraits from "../Traits/DiscreteTimeTraits";
-import sortedIndexOf from "lodash/sortedIndexOf";
+import { action, computed, observable, onBecomeObserved } from "mobx";
 import Ellipsoid from "terriajs-cesium/Source/Core/Ellipsoid";
+import JulianDate from "terriajs-cesium/Source/Core/JulianDate";
 import CesiumMath from "terriajs-cesium/Source/Core/Math";
+import Entity from "terriajs-cesium/Source/DataSources/Entity";
+import Constructor from "../Core/Constructor";
 import filterOutUndefined from "../Core/filterOutUndefined";
+import LatLonHeight from "../Core/LatLonHeight";
+import runLater from "../Core/runLater";
+import { ProviderCoords, ProviderCoordsMap } from "../Map/PickedFeatures";
 import CommonStrata from "../Models/CommonStrata";
-import { computed, action, observable, onBecomeObserved } from "mobx";
-import TimeVarying from "./TimeVarying";
+import createStratumInstance from "../Models/createStratumInstance";
 import Mappable, { ImageryParts } from "../Models/Mappable";
+import Model from "../Models/Model";
+import DiscretelyTimeVaryingTraits from "../Traits/DiscretelyTimeVaryingTraits";
 import MappableTraits from "../Traits/MappableTraits";
-import runLater from "../Core/runLater";
-import LatLonHeight from "../Core/LatLonHeight";
+import TimeFilterTraits, {
+  TimeFilterCoordinates
+} from "../Traits/TimeFilterTraits";
+import DiscretelyTimeVaryingMixin from "./DiscretelyTimeVaryingMixin";
+import TimeVarying from "./TimeVarying";
 
 type MixinModel = Model<
   TimeFilterTraits & DiscretelyTimeVaryingTraits & MappableTraits
 > &
-  TimeVarying;
+  DiscretelyTimeVaryingMixin.Instance;
 
 /**
  * A Mixin for filtering the dates for which imagery is available at a location
- * picked by the user.
+ * picked by the user. This Mixin requires `DiscretelyTimeVaryingMixin` to be present.
```

Comment:
Rather than one Mixin requiring another to be present I think it is better to have TimeFilterMixin mix in DiscretelyTimeVaryingMixin and itself when called. Then TimeFilterTraits would also include DiscretelyTimeVaryingTraits. Unless that isn't technically possible or poses problems somewhere.